apps/studio/src/pages/sites/[siteId]/pages/[pageId]/index.tsx

```diff
@@ -117,148 +122,169 @@ const PageEditingView = ({ page, permalink, siteId }: PageEditingViewProps) => {
   )
 }
 
-const pageSettingsSchema = z.object({
-  title: pageTitleSchema,
-  meta: z.string().optional(),
-  permalink: permalinkSchema,
-  noIndex: z.boolean().optional(),
-})
+const THREE_SECONDS = 3000
+const SUCCESS_TOAST_ID = "save-page-settings-success"
 
 interface PageSettingsProps {
   permalink: RouterOutput["page"]["readPageAndBlob"]["permalink"]
-  page: RouterOutput["page"]["readPageAndBlob"]["content"]
+  title: RouterOutput["page"]["readPageAndBlob"]["title"]
+  type: RouterOutput["page"]["readPageAndBlob"]["type"]
 }
 const PageSettings = ({
   permalink: originalPermalink,
-  page,
+  type,
+  title: originalTitle,
 }: PageSettingsProps) => {
-  const { register, setValue, getFieldState, watch, control } = useZodForm({
-    schema: pageSettingsSchema,
+  const { pageId, siteId } = useQueryParse(editPageSchema)
+  const { register, watch, control, handleSubmit } = useZodForm({
+    schema: pageSettingsSchema.omit({ pageId: true, siteId: true }),
     defaultValues: {
-      title: page.page.title || "",
+      title: originalTitle || "",
       permalink: originalPermalink || "",
     },
   })
 
   const [title, permalink] = watch(["title", "permalink"])
 
-  useEffect(() => {
-    const permalinkFieldState = getFieldState("permalink")
-    // This allows the syncing to happen only when the page title is not dirty
-    // Dirty means user has changed the value AND the value is not the same as the default value of "".
-    // Once the value has been cleared, dirty state will reset.
-    if (!permalinkFieldState.isDirty) {
-      setValue("permalink", generateResourceUrl(title), {
-        shouldValidate: !!title,
+  const toast = useToast({ duration: THREE_SECONDS, isClosable: true })
+  const utils = trpc.useUtils()
+
+  const updatePageSettingsMutation = trpc.page.updatePageSettings.useMutation({
+    onSuccess: async () => {
+      // TODO: we should use a specialised query for this rather than the general one that retrives the page and the blob
+      await utils.page.readPageAndBlob.invalidate()
+      await utils.page.readPage.invalidate()
+      if (!toast.isActive(SUCCESS_TOAST_ID)) {
+        toast({
+          id: "save-page-settings-success",
```

Comment:
should you use `SUCCESS_TOAST_ID` here too? Since you already have the constant

Reply:
ah oops, i missed this one! thanks for teh catch